server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java

```diff
@@ -1096,6 +1096,26 @@ protected void runInContext() {
             }
         }
 
+        private void syncBackupMetrics(final BackupProvider backupProvider, final Map<VirtualMachine, Backup.Metric> metrics) {
+            for (final VirtualMachine vm : metrics.keySet()) {
+                try {
+                    final Backup.Metric metric = metrics.get(vm);
+                    if (metric != null) {
+                        LOG.debug(String.format("Trying to sync backups of VM [%s] using backup provider [%s].", vm.getUuid(), backupProvider.getName()));
+                        // Sync out-of-band backups
+                        backupProvider.syncBackups(vm, metric);
+                        // Emit a usage event, update usage metric for the VM by the usage server
+                        UsageEventUtils.publishUsageEvent(EventTypes.EVENT_VM_BACKUP_USAGE_METRIC, vm.getAccountId(),
+                                vm.getDataCenterId(), vm.getId(), "Backup-" + vm.getHostName() + "-" + vm.getUuid(),
+                                vm.getBackupOfferingId(), null, metric.getBackupSize(), metric.getDataSize(),
+                                Backup.class.getSimpleName(), vm.getUuid());
+                    }
+                } catch (final Exception e) {
+                    LOG.error(String.format("Failed to sync backup usage metrics and out-of-band backups of VM [%s] due to: [%s].", vm.getUuid(), e.getMessage()), e);
+                }
```

Comment:
for readability the try block can be a separate method.

Reply:
Done, thanks for the suggestion.